sdk/core/az_core/src/az_http_policy.c

```diff
@@ -70,11 +60,16 @@ AZ_NODISCARD az_result az_http_pipeline_policy_credential(
     _az_http_request* p_request,
     az_http_response* p_response)
 {
-  if (p_data != AZ_CREDENTIAL_ANONYMOUS)
+  _az_credential* const credential = (_az_credential*)p_data;
```

Comment:
I think p_data should be p_options for consistency.
Or, if we could make the 2nd parameter be the right type and the right name, that would be better; we won't have to cast in the function (like this line 63).

Reply:
Done :)